server/src/main/java/org/jetbrains/bsp/bazel/server/sync/model/Project.kt

```diff
@@ -6,7 +6,8 @@ import java.net.URI
 data class Project(
     val workspaceRoot: URI,
     val modules: List<Module>,
-    val sourceToTarget: Map<URI, Label>
+    val sourceToTarget: Map<URI, Label>,
+    val rustExternalModules: List<Module> = emptyList()
```

Comment:
hmm, this looks like a very specific thing thrown among very general things

Reply:
Yeah, I think it was added because these modules shouldn't be seen as `buildTargets`. I think I managed to remove it and instead added a field to `RustModule` -> `isExternalModule` to mark these modules and make them filterable. Now these `rustExternalModules` are mixed into `modules` in the `Project`, so I made an additional function to filter them. I believe that the only place that we should use this filter is in `BspProjectMapper` while establishing `workspaceTargets`, but maybe @abrams27 knows some other place